common/djangoapps/entitlements/tasks.py

```diff
@@ -62,3 +72,49 @@ def expire_old_entitlements(self, start, end, logid='...'):
         raise self.retry(exc=exc, countdown=countdown, max_retries=MAX_RETRIES)
 
     LOGGER.info('Successfully completed the task expire_old_entitlements after examining %d entries [%s]', entitlements.count(), logid)  # lint-amnesty, pylint: disable=line-too-long
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def expire_and_create_entitlements(self, no_of_entitlements):
+    """
+    This task is designed to be called to process and expire bundle of entitlements
+    that are older than one year on in exceptional case 18 months.
+
+    Args:
+        None
+
+    Returns:
+        None
+
+    """
```

Comment:
[docs] Clarify that this task is meant to create new entitlements.

```suggestion
    """
    Expire entitlements older than one year.
    
    Exception: if the entitlement is for a course in a list of exceptional courses,
    expire those entitlements if they're older than 18 months instead.
     
    Then create a copy of the expired entitlement to renew it for another year
    / 18 months.

    Args:
        no_of_entitlements (int): Limit the operation to this number of entitlements.

    Returns:
        None

    """
```

Reply:
Done by @zubair-ce07 in 889ae7bc9c667cf61e6eed7d8af60c91e61ae381.